x-pack/plugins/triggers_actions_ui/public/application/sections/alert_details/components/alerts_route.tsx

```diff
@@ -30,17 +31,48 @@ export const AlertsRoute: React.FunctionComponent<WithAlertSummaryProps> = ({
   readOnly,
   requestRefresh,
   loadAlertSummary: loadAlertSummary,
+  refreshToken,
 }) => {
   const {
     notifications: { toasts },
   } = useKibana().services;
 
   const [alertSummary, setAlertSummary] = useState<AlertSummary | null>(null);
+  const [numberOfExecutions, setNumberOfExecutions] = useState(60);
+  const [isLoadingChart, setIsLoadingChart] = useState(true);
+  const ruleID = useRef<string | null>(null);
+  const refreshTokenRef = useRef(refreshToken);
+
+  const getAlertSummaryWithLoadingState = useCallback(
+    async (executions: number = numberOfExecutions) => {
+      setIsLoadingChart(true);
+      await getAlertSummary(ruleID.current!, loadAlertSummary, setAlertSummary, toasts, executions);
+      setIsLoadingChart(false);
+    },
+    [setIsLoadingChart, ruleID, loadAlertSummary, setAlertSummary, toasts, numberOfExecutions]
+  );
+
+  useEffect(() => {
+    if (ruleID.current !== rule.id) {
+      ruleID.current = rule.id;
+      getAlertSummaryWithLoadingState();
+    }
+  }, [rule, ruleID, getAlertSummaryWithLoadingState]);
 
   useEffect(() => {
-    getAlertSummary(rule.id, loadAlertSummary, setAlertSummary, toasts);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [rule]);
+    if (refreshTokenRef.current !== refreshToken) {
```

Comment:
Is this `refreshToken` logic to avoid spam clicking the refresh button? or something else?